### DISCUSSION TITLE
what is the purpose of this line in lazy.lua?

### DISCUSSION THREAD

**Discussion by alivault** [LazyVim/LazyVim#2650] [General]
https://github.com/LazyVim/starter/blob/741ff3aa70336abb6c76ee4c49815ae589a1b852/lua/config/lazy.lua#L29

in lazy.lua there is this line:
`install = { colorscheme = { "tokyonight", "habamax" } },`

Commenting it out seems to have no effect, as tokyonight and habamax stay available in the colorscheme switcher upon reload.

Also I see tokyonight is in `lazy-lock.json` but habamax is not. Yet they both stay available in the colorscheme switcher 🤔

Anyway, I like tokyonight-night so I just set it in a `colorscheme.lua` file, while also adding more themes I can switch between later:

```
return {
  { "Mofiqul/vscode.nvim" },
  { "rebelot/kanagawa.nvim" },
  { "catppuccin/nvim" },
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "tokyonight-night",
    },
  },
}
```

Is there a better way to manage these themes? Is it possible to use the colorscheme switcher to update the config file for me, or is it just for temporary preview?

**Comment by Demianeen** [comment]
Have you tried to use "clean" action in lazy panel?

**Comment by skoch13** [comment]
This file you referred to used by [lazy.nvim](https://github.com/folke/lazy.nvim) configuration: 'try to load one of these color schemes when starting an installation during startup.'

Next:

> Anyway, I like tokyonight-night, so I just set it in a colorscheme.lua file.

Another way to approach this would be not changing the `LazyVim` spec, but configuring the `tokyonight` theme as follows:

```lua
{
  "folke/tokyonight.nvim",
  opts = { style = "night" }
}
```

> Is it possible to use the colorscheme switcher to update the config file for me?

The docs suggest updating the LazyVim spec to make a permanent change, while `<leader>uC` changes the color scheme for the current session.

Please also don't forget to add `lazy = false` for those color schemes you want to always be listed in the colorscheme switcher.

Hope that helps.

**Reply by alivault** [reply]
Thanks for the explanation.

This seems to be working. Is this what you meant?

```lua
-- colorscheme.lua
return {
  { "Mofiqul/vscode.nvim", lazy = false },
  { "rebelot/kanagawa.nvim", lazy = false },
  { "catppuccin/nvim", lazy = false },
  {
    "folke/tokyonight.nvim",
    opts = { style = "night" },
  },
}
```

**Reply by dpetka2001** [reply]
You might want to take a look at this [Discussion](https://github.com/folke/lazy.nvim/discussions/1167) in `lazy.nvim` repo. It provides a way to lazy load your colorschemes and then load them only when you invoke `Telescope colorscheme`, as well as a way to filter out builtin Neovim colorschemes and only show the ones you'd like, if you wanted something like this as an option.

**Reply by alivault** [reply]
thanks @dpetka2001, I was able to set up my colorschemes to lazy load but still be available in the picker, thanks to your suggestion. Code here: https://github.com/folke/lazy.nvim/discussions/1167#discussioncomment-8726381
